### PR TITLE
Add support for more control over Android PopBackStackImmediate on fragments

### PR DESCRIPTION
**Pull request by Plac3hold3r**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Feature

### :arrow_heading_down: What is the current behavior?
Defaults to inclusive and the FragmentJavaName. If you have two of the same fragments called after each other both would get closed with a `NavigationService.Close(this)`.

### :new: What is the new behavior (if this is a feature change)?
Users will have more control over how they want the close of a fragment to function.

### :boom: Does this PR introduce a breaking change?
No

### :bug: Recommendations for testing
See sample `Fragment Close Example` in Plaground. You can switch between the different fragment attributes, [default](https://github.com/MvvmCross/MvvmCross/pull/3159/files#diff-4743e86237e08bf75fe3bb341e3da764R13) or [pop top only](https://github.com/MvvmCross/MvvmCross/pull/3159/files#diff-4743e86237e08bf75fe3bb341e3da764R14). 

### :memo: Links to relevant issues/docs
Updated documentation.

### :thinking: Checklist before submitting

- [X] All projects build
- [X] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [X] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contributing/mvvmcross-docs-style-guide))
- [X] Rebased onto current develop

### CODE REVIEW

**Comment by Cheesebaron** [comment]
Very nice Pull Request with docs and all. Good job!

File: MvvmCross.Android.Support/V7.AppCompat/MvxAppCompatViewPresenter.cs [hunk 1]
```diff
@@ -641,12 +641,16 @@ protected override Task<bool> CloseFragment(IMvxViewModel viewModel, MvxFragment
 
             if (fragmentManager.BackStackEntryCount > 0)
             {
-                fragmentManager.PopBackStackImmediate(fragmentName, 1);
+                var popBackStackFragmentName = fragmentAttribute.PopBackStackImmediateName == ""
```
[the remark this answers is not in the document]
Comment (Plac3hold3r) [reply MvvmCross.Android.Support/V7.AppCompat/MvxAppCompatViewPresenter.cs]: We would want to allow for null, as a null is a valid parameter to [`PopBackStackImmediate()`](https://developer.android.com/reference/android/app/FragmentManager.html#popBackStack(java.lang.String,%20int)). It allows for popping the current top fragment. The empty string, default state, I have mapped to the current behaviour where the Java fragment name is used.

[the remark this answers is not in the document]
Comment (Cheesebaron) [reply MvvmCross.Android.Support/V7.AppCompat/MvxAppCompatViewPresenter.cs]: RIght, OK. I only glanced over it. What about if someone adds whitespace only for the name, should it be trimmed?

[the remark this answers is not in the document]
Comment (Plac3hold3r) [reply MvvmCross.Android.Support/V7.AppCompat/MvxAppCompatViewPresenter.cs]: That's a good point. I'm not sure if whitespace is a valid character in naming a fragment. I'll test it out. If not, then I can add the additional check to trim the whitespace.

**Comment by Plac3hold3r** [comment]
> Very nice Pull Request with docs and all. Good job!

Thanks :)

File: MvvmCross.Android.Support/V7.AppCompat/MvxAppCompatViewPresenter.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Plac3hold3r) [reply MvvmCross.Android.Support/V7.AppCompat/MvxAppCompatViewPresenter.cs]: @Cheesebaron I checked and a whitespace parameter is not allowed. According to the Android [docs](https://developer.android.com/reference/android/app/Fragment.html#instantiate(android.content.Context,%20java.lang.String,%20android.os.Bundle)) the `fname` must be a class name. As `PopBackStackImmediate` also uses this name as well, it must also be a valid class name or null. So I have added a trim to the `PopBackStackImmediateName` before evaluating.

**Pull request merged**